airbyte-integrations/connectors/source-notion/source_notion/streams.py

```diff
@@ -121,6 +132,12 @@ def next_page_token(
             return {"next_cursor": next_cursor}
 
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+
+        # If page_size has been reduced after encountering a 504 Gateway Timeout error,
+        # we increase it back to the default of 100 once a success response is achieved, for the following API calls.
+        if response.status_code == 200 and self.page_size != 100:
```

Comment:
same thing, please move to should_rerty to align with SRP principle

Reply:
Done and done! All logic to reduce/reset page_size has been moved to should_retry method